### DISCUSSION TITLE
Migration to v7

### DISCUSSION THREAD

**Discussion by czkoudy** [react-hook-form/react-hook-form#5254] [Q&A]
Hey, 

I had this in version 6
```
 ref={(e) => {
                 monitoring.current = e;
                 register(e);
                 }}
```

how do i transform this to v7?

**Comment by czkoudy** [comment]
Hi, i added a codesandbox with all my code. If you hit submit you will get console.log with true or false. However this will always be what is default. At the moment is set to true so it will always print true regardless what toggleswitch ise set to :-( any idea?

Please have a look at [Codesandbox](https://codesandbox.io/s/empty-sea-cxbzw)

**Comment by bluebill1049** [comment]
maybe this section will help: https://react-hook-form.com/faqs#Howtosharerefusage

**Reply by czkoudy** [reply]
Thanks for the quick reply @bluebill1049 I have had a look at that and used it in my codesandbox but still no luck :-(

**Reply by czkoudy** [reply]
and if you look at this [codesandbox](https://codesandbox.io/s/jovial-microservice-1kvem?file=/src/App.js) it works fine in v6
